### PR TITLE
Add `RuntimeMaxSec=` support to scope units (time-limited login sessions)

### PR DESCRIPTION
**Pull request by pwithnall**
This is an implementation of #12035, which follows Lennart’s recommended approach:
 * Add `RuntimeMaxSec=` support to scope units
 * Pass through the `systemd.runtime_max_sec` PAM data in `pam_systemd.so` to set that scope property
 * There are some refactoring commits first

Tested the `RuntimeMaxSec=` support using:
```
# systemd-run --scope --property=RuntimeMaxSec=5s sleep 10
Running scope as unit: run-r00615ed439d044b7b3f9930965c6cd52.scope
Terminated
[root@image ~]# echo $?
143
[root@image ~]# journalctl -b | tail -n 5
Aug 28 18:00:11 image systemd[1]: run-r50ef4e09714844329203c247438a51a4.scope: Scope reached runtime time limit. Stopping.
Aug 28 18:00:11 image systemd[1]: run-r50ef4e09714844329203c247438a51a4.scope: Failed with result 'timeout'.
Aug 28 18:00:55 image systemd[1]: Started /usr/bin/sleep 10.
Aug 28 18:01:00 image systemd[1]: run-r00615ed439d044b7b3f9930965c6cd52.scope: Scope reached runtime time limit. Stopping.
Aug 28 18:01:00 image systemd[1]: run-r00615ed439d044b7b3f9930965c6cd52.scope: Failed with result 'timeout'.
```

and tested the `pam_systemd.so` changes by temporarily hard-coding `runtime_max_sec = "60";` in `pam_systemd.c` and then logging in to a session:
```
# cat /run/systemd/transient/session-8.scope
# This is a transient unit file, created programmatically via the systemd API. Do not edit.
[Scope]
Slice=user-0.slice

[Unit]
Description=Session 8 of user root
Wants=user-runtime-dir@0.service
Wants=user@0.service
After=systemd-logind.service
After=systemd-user-sessions.service
After=user-runtime-dir@0.service
After=user@0.service
RequiresMountsFor=/root

[Scope]
SendSIGHUP=yes
TasksMax=infinity
RuntimeMaxSec=1min
```
(note the `RuntimeMaxSec=` at the bottom)

The session was successfully and automatically ended 1 minute after logging in.

### CODE REVIEW

File: src/login/pam_systemd.c [hunk 1]
```diff
@@ -279,6 +279,27 @@ static int append_session_memory_max(pam_handle_t *handle, sd_bus_message *m, co
...
+
+        /* No need to parse "infinity" here, it will be set by default later in scope_init() */
+        if (isempty(limit) || streq(limit, "infinity"))
+                return 0;
+
+        r = parse_sec(limit, &val);
+        if (r >= 0) {
+                r = sd_bus_message_append(m, "(sv)", "RuntimeMaxUSec", "t", (uint64_t) val);
+                if (r < 0) {
+                        pam_syslog(handle, LOG_ERR, "Failed to append to bus message: %s", strerror(-r));
```
Comment (yuwata) [inline src/login/pam_systemd.c]: `strerror(-r)` -> `strerror_safe(r)`

Comment (pwithnall) [reply src/login/pam_systemd.c]: Fixed, thanks.

**Comment by yuwata** [comment]
LGTM. Adding test for the functionality is welcome.

**Comment by pwithnall** [comment]
I added a test for `RuntimeMaxSec=` to the `TEST-03-JOBS` integration test, but have not been able to run it (or any of the integration tests) locally; hopefully the CI does.

File: src/shared/bus-unit-util.c [hunk 2]
```diff
@@ -1401,6 +1401,14 @@ static int bus_append_path_property(sd_bus_message *m, const char *field, const
         return 0;
 }
 
+static int bus_append_scope_property(sd_bus_message *m, const char *field, const char *eq) {
+        if (streq(field, "TimeoutStopSec"))
+
```
Comment (keszybz) [inline src/shared/bus-unit-util.c:1410]: This empty newline here is not necessary... but I see that this matches the surrounding code. OK, let's keep it this way for now.

File: man/pam_systemd.xml [hunk 3]
```diff
@@ -263,6 +263,12 @@
 
         <listitem><para>Sets unit <varname>IOWeight=</varname>.</para></listitem>
       </varlistentry>
+
```
Comment (keszybz) [inline man/pam_systemd.xml:266]: Hmm, the commit message says "Usec", but it's really "USec".

File: test/TEST-03-JOBS/test-jobs.sh [hunk 4]
```diff
@@ -84,4 +84,13 @@ END_SEC=$(date -u '+%s')
 ELAPSED=$(($END_SEC-$START_SEC))
 [[ "$ELAPSED" -ge 5 ]] && [[ "$ELAPSED" -le 7 ]] || exit 1
 
+# Test time-limited scopes
+START_SEC=$(date -u '+%s')
+systemd-run --scope --property=RuntimeMaxSec=3s sleep 10
```
Comment (keszybz) [inline test/TEST-03-JOBS/test-jobs.sh:90]: This will cause the test to fail because we're running under `set -e`.

**Pull request merged**